djot/inline.lua

```diff
@@ -330,14 +329,14 @@ Parser.matchers = {
             self:add_match(opener[1], opener[2], "+linktext")
             self:add_match(opener[4], opener[4], "-linktext")
           end
-          self:add_match(startdest, startdest, "+destination")
-          self:add_match(enddest, enddest, "-destination")
+          self:add_match(opener[5], opener[5], "+destination")
+          self:add_match(pos, pos, "-destination")
           self.destination = false
           -- convert all matches to str
           self:str_matches(opener[5] + 1, pos - 1)
           -- remove from openers
-          self:clear_openers(opener[2], pos)
-          return enddest + 1
+          self:clear_openers(opener[1], pos)
```

Comment:
And this one I am not really sure about: this definitely different between the two code paths, but I am not yet able to come up with the test here, as I don't understand openers enough yet.

Reply:
Ok, now I am sure about this:  we do want to clear all openers from the start of the current one. As it happens, as our openners are single-char-wide and we use closed intervals, this is the same. But, if we were to use a multichar openner here, this would break.